packages/common/src/global-grid-options.ts

```diff
@@ -279,5 +279,10 @@ function pickerHeaderColumnValueExtractor(column: Column, gridOptions?: GridOpti
   if (headerGroup) {
     return headerGroup + columnGroupSeparator + column.name;
   }
+
+  if (column?.name instanceof HTMLElement || column?.name instanceof DocumentFragment) {
+    return column.name.textContent ?? '';
+  }
+
```

Comment:
so I had a very similar approach but I think it would be better to go with mine instead, because with yours, you forgot to deal with the line 280 

```suggestion
function pickerHeaderColumnValueExtractor(column: Column, gridOptions?: GridOption) {
  let colName = column?.name ?? '';
  if (colName instanceof HTMLElement || colName instanceof DocumentFragment) {
    colName = colName.textContent || '';
  }
  const headerGroup = column?.columnGroup || '';
  const columnGroupSeparator = gridOptions?.columnGroupSeparator ?? ' - ';
  if (headerGroup) {
    return headerGroup + columnGroupSeparator + colName;
  }
  return colName;
}
```